### PR TITLE
Add Express Middleware for exporting Prometheus Metrics

### PR DESCRIPTION
**Pull request by avatarneil**
## 📚 Purpose
Currently, mds-core doesn't export many meaningful operational metrics per-endpoint. This adds the express-prom-bundle middleware, which will export a slew of new metrics that Prometheus can scrape. Once Prometheus has the metrics, we can set up alerting and make pretty dashboards in Grafana!

## 📦 Impacts:
Anything that uses the mds-api-server

### CODE REVIEW

File: packages/mds-api-server/index.ts [hunk 1]
```diff
@@ -12,6 +12,9 @@ import {
   UserEmailClaim,
   JurisdictionsClaim
 } from '@mds-core/mds-api-authorizer'
+import promBundle from 'express-prom-bundle'
+
+const metricsMiddleware = promBundle({includeMethod: true, includePath: true, includeUp: true})
```
Comment (drtyh2o) [inline packages/mds-api-server/index.ts:17]: I may have suggested `prometheus` and `prometheusMiddleware` ... Outside of this PR, it's not obvious what `prom` is.

**Comment by invertigo** [comment]
Two questions:

@drtyh2o This middleware exposes a new path - `/metrics` for consumption by prometheus.  Will this collide with any path we're currently using or planning on implementing?  It would be pretty straightforward to change the path to something like `/prometheus` which would have less risk of a collision.

@avatarneil does this middleware allow for adding additional custom metrics (more code instrumentation) or labels?  Seems like if we can add a `provider_id=` label to the existing metrics, that would be immediately useful.

**Comment by drtyh2o** [comment]
@invertigo I don't think it will cause any collisions presently. The Metrics API uses `/metrics` as its `pathPrefix` and hosts its routes at the root so technically `/metrics/metrics` is available, but if it's easy to change then we ought to just to avoid confusion. I'd rather reserve _metrics_ to mean MDS/AGTM metrics and not conflate with prometheus.

**Comment by avatarneil** [comment]
I'll switcheroo the path to `/prometheus`. @invertigo I'm still not totally sure about custom metrics, if this doesn't support it, we may want to switch to the other (less used) middleware that I found last week. But I'd like to get this out the door soon

**Comment by drtyh2o** [comment]
`mds-compliance` API tests are failing with
```
Error: A metric with the name http_request_duration_seconds has already been registered.
```
`mds-compliance` is the only package having an issue. I am wondering if this is somehow related to how it runs both agency/compliance during tests @avatarneil

**Comment by drtyh2o** [comment]
In addition to the above issue which is due to `express-prom-bundle` using the `prom-client` global registry with no support for prom-client's `registry` option. It also seems that the `metricsPath` only accepts a string so setting it to `/prometheus` will not place that route under the `PATH_PREFIX` that all of the other routes are on. So, endpoint routes will be:
```
/agency/health
/agency/vehicles
/prometheus
```
The `metricsPath` won't accept the output of `pathsFor` which binds a route handler to multiple paths. I wonder if that's a feature that we've outgrown. I wonder if we could change `pathsFor` to `pathFor` that would just add the PATH_PREFIX.

The collection of these smaller issues makes me wonder if we'd be better off just using `prom-client` directly and implementing our own route in `mds-api-server` where we could use separate registries for each API.

File: packages/mds-api-server/index.ts [hunk 2]
```diff
@@ -287,7 +285,7 @@ export const ApiServer = (
 
   // Middleware
   app.use(
-    PrometheusMiddlware(),
+    PrometheusMiddlware,
```
Comment (drtyh2o) [inline packages/mds-api-server/index.ts:288]: I get it, but boo.

**Comment by avatarneil** [comment]
@invertigo I think we may be able to harness the `customLabel` and `transformLabel` properties to extract `provider_id`! Can add that functionality in a subsequent PR.

**Comment by avatarneil** [comment]
@invertigo Before we merge this, can we deploy to sputnik and verify prometheus is working as expected?

**Pull request merged**